paddle/framework/lod_tensor_impl.h

```diff
@@ -14,27 +14,47 @@
 
 #pragma once
 
-#include "paddle/framework/details/lod_tensor.h"
-
 namespace paddle {
 namespace framework {
 
+namespace detail {
+
+using LOD = LODTensor::LOD;
+
+/*
+ * Slice levels from LOD.
+ *
+ * @lod: LOD to slice.
+ * @level_begin: level to begin slice.
+ * @level_end: level to end slice.
+ */
+LOD SliceLOD(const LODTensor::LOD &lod, size_t level_begin, size_t level_end);
+/*
+ * Slice elements from a level of LOD.
+ *
+ * @lod: LOD to slice.
+ * @level: which level to slice.
+ * @elem_begin: element's index to begin slice.
+ * @elem_end: element's index to end slice.
+ */
+LOD SliceLOD(const LODTensor::LOD &lod, size_t level, size_t elem_begin,
+             size_t elem_end);
+
+}  // namespace detail
+
 template <typename T>
-LODTensor LODTensor::SliceCopied(size_t level_begin, size_t level_end,
-                                 const platform::Place &dst_place) const {
-  PADDLE_ENFORCE(HasLOD(), "has no LOD info, can't be sliced.");
-  auto new_lod = details::SliceLOD(*lod_start_pos_, level_begin, level_end);
-  auto new_tensor = std::make_shared<Tensor>();
-  new_tensor->CopyFrom<T>(*tensor_, dst_place);
-
-  return LODTensor(new_tensor, new_lod);
+LODTensor LODTensor::SliceLevels(size_t level_begin, size_t level_end) const {
+  auto new_lod = detail::SliceLOD(lod_start_pos_, level_begin, level_end);
```

Comment:
It seems that if we can fix https://github.com/PaddlePaddle/Paddle/issues/3376, then L47 to L52 could be merged into one line?

```cpp
return LODTensor(lod_.Slice(level_begin, level_end)).ShareDataWith<T>(*this);
```